Mathlib/Data/List/Lex.lean

```diff
@@ -61,6 +61,19 @@ theorem not_nil_right (r : α → α → Prop) (l : List α) : ¬Lex r l [] :=
   fun.
 #align list.lex.not_nil_right List.Lex.not_nil_right
 
+theorem nil_left_or_nil_eq {r : α → α → Prop} (l : List α) : List.Lex r [] l ∨ [] = l := by
```

Comment:
Can you add a version assuming `IsRefl r` that says `List.Lex r [] l`?

Reply:
That result is not true. `List.Lex` is not reflexive even though `r` is. Indeed, we have: `List.Lex.not_nil_right : ¬List.Lex r l []`, in particular `¬List.Lex r [] []`